pennylane/beta/queuing/measure.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=protected-access
+"""
+This module contains the functions for computing different types of measurement
+outcomes from quantum observables - expectation values, variances of expectations,
+and measurement samples using AnnotatedQueues.
+"""
+import collections
+
+import pennylane as qml
+from pennylane.operation import Expectation, Observable, Probability, Sample, Variance
+from pennylane.ops import Identity
+from pennylane.qnodes import QuantumFunctionError
+
+MeasurementProcess = collections.namedtuple("MeasurementProcess", ["return_type"])
+"""NamedTuple: A namedtuple that contains the return_type of the circuit and
+whose instance can be queried by id."""
```

Comment:
Turned out that the [measurement types in PennyLane are defined as Enum](https://github.com/PennyLaneAI/pennylane/blob/38ffcb4bab6fc63f3a6e3c4e1eeb364a8b78ed7d/pennylane/operation.py#L146). That is they are singleton and cannot be differentiated if put in the queue multiple times. Therefore, a need for a wrapper around them would be required, one that can be instantiated. Once instantiated, the id of an instance could be queried when looking at the queue of an `AnnotatedQueue`.

Worth noting, that we would like an object that merely stores a single piece of information (`return_type`). [After looking into possibilities](https://stackoverflow.com/questions/3357581/using-a-class-as-a-data-container), it looked like using a `namedtuple` could be fitting here (thank you @josh146  with the help in this!).

It would be a smooth transition to make `MeasurementProcess` a class, should we want to add more members and/or methods to it.